### PR TITLE
fix/stripe-customer-sync

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

Takes care of keeping in sync user data with stripe customer data

Fixes # (issue)
**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Change email or username on settings page then look up on stripe for customer's metadata and they should be the same as our DB.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/server/routers/viewer.tsx [hunk 1]
```diff
@@ -707,12 +709,30 @@ const loggedInViewerRouter = createProtectedRouter()
...
+        select: {
+          id: true,
+          username: true,
+          email: true,
+          metadata: true,
+        },
       });
+
+      // Notify stripe about the change
+      if (updatedUser && updatedUser.metadata && hasKeyInMetadata(updatedUser, "stripeCustomerId")) {
```
Comment (alannnc) [inline apps/web/server/routers/viewer.tsx:726]: @zomars Should I add the same flag to know if we are on self hosted? And don't execute any stripe code.

Comment (zomars) [reply apps/web/server/routers/viewer.tsx:726]: Some self-hosted users will use Stripe Connect, If this is only for premium usernames then we can skip it

**Review by zomars — APPROVED** [review]

File: apps/web/server/routers/viewer.tsx (hunk 1 above)
Comment (alannnc) [reply apps/web/server/routers/viewer.tsx:726]: Let me get this right, connect can be for self hosted users, but this it's only involves our user plan handling so I should add flag here for `IS_SELF_HOSTED`.

Comment (zomars) [reply apps/web/server/routers/viewer.tsx:723]: Nevermind seems like is good to go 👍🏽

**Pull request merged**